Sofa/Component/Engine/Transform/src/sofa/component/engine/transform/DilateEngine.inl

```diff
@@ -92,13 +81,21 @@ void DilateEngine<DataTypes>::doUpdate()
     const Real minThickness = d_minThickness.getValue();
 
     WriteOnlyAccessor<Data<VecCoord> > out = d_outputX;
+    WriteOnlyAccessor<Data<VecCoord> > normals = d_normals;
 
     const int nbp = in.size();
     const int nbt = triangles.size();
     const int nbq = quads.size();
 
-    WriteOnlyAccessor<Data<VecCoord> > normals = d_normals;
+    if(nbt == 0)
+        msg_warning() << "No input mesh (no triangle)";
+    if(nbq == 0)
+        msg_warning() << "No input mesh (no quad)";
```

Comment:
Let's say you have a mesh with only triangles. It means you will have a warning because you don't have quads. Do we want that?